geopandas/tools/tests/test_clip.py

```diff
@@ -450,3 +450,42 @@ def test_clip_single_multipoly_no_extra_geoms(
     multi = buffered_locations.dissolve(by="type").reset_index()
     clipped = clip(multi, masks)
     assert clipped.geom_type[0] == "Polygon"
+
+
+def empty_mask(frame=False, mixed=False):
+    x = [np.nan] * 3
+    y = np.arange(3) if mixed else x
+    mask = GeoSeries.from_xy(x, y, crs="EPSG:3857").buffer(1)
+    if frame:
+        mask = mask.to_frame()
+    return mask
+
+
+@pytest.mark.filterwarnings("ignore:All-NaN slice encountered")
+@pytest.mark.parametrize(
+    "mask",
+    [
+        Point(0, 0).buffer(0),
+        Point(np.nan, np.nan).buffer(1),
+        Point(1, np.nan).buffer(1),
```

Comment:
I think we can reduce a bit the parametrization. All three above create an empty Polygon, so for testing our code, this is actually all the same. Given my other comment about compat with older GEOS, I think this can be replaced with a single `Polygon()` (or the `Point(0, 0).buffer(0)` will actually also work)